### PR TITLE
fix(attention): prepare sparse MLA through its module owner

### PR DESCRIPTION
**Pull request by voipmonitor**
## Behavior

Make cache-dependent B12X sparse MLA preparation reachable through the owning `MLAAttention` module, including when the optional BF16 query projection is absent. The backend implementation is not an `nn.Module`, so a traversal of modules alone misses its registered provider.

Preparation borrows caller-owned workspace using the prepared state's public `scratch_specs()` contract. Exact-row plans no longer each retain a private scratch allocation. For GLM DCP1, the provider also declares pooled-index expansion before kernel resolution is frozen.

Normalize backend results at the vLLM boundary: decode can return an output tensor, while extend can return `(output, LSE)`. Return LSE only when the consumer requires it; reject a missing required LSE. Attention arithmetic and DCP reduction behavior are unchanged.

## Dependencies and compatibility

Requires local-inference-lab/b12x#379 for `plan_pooled_selection`. A missing preparation API fails explicitly during startup. No environment switch is added.

This complements #772, the consolidated GLM startup alternative to #761. #772 owns mHC/KDA operands, indexer geometry and unique identities for both explicitly named and anonymous indexers. Preparation discovery, caller-owned sparse scratch, physical-selection plans and result normalization remain here. The branches merge without manual conflict resolution; do not merge both #761 and #772.

#739 also touches sparse scratch allocation. Preserve the public `scratch_specs()` binding and shared caller-owned workspace when composing those changes; do not restore per-plan private scratch allocations.

## Validation

Status: implemented and qualified in the GLM NVFP4 TP4/DCP1 DFlash2 K7 composition. Ten focused tests pass on the SM120 runtime, with GPU tests enabled:

```sh
B12X_GLM53_GPU_TEST=1 python -m pytest --noconftest -q \
  tests/v1/attention/test_b12x_sparse_mla_api.py \
  tests/models/test_glm5next_pooled_indexer.py \
  -k 'declares_plans_once_and_prepares_in_place or forward_normalizes_decode_and_extend_results or preparation_borrows_declared_scratch_shape or discovers_backend_and_optional_query_plans or physical_selection_prepares_before_resolution_freeze'
```

Coverage: discovery with/without query projection, DCP1 versus DCP4 selection preparation, exact scratch shape/dtype and reuse, decode/extend return contracts, and prepared expansion for 1/4/32 live rows without resolving another kernel. #772 carries the anonymous-indexer identity regression.

The independently serving-qualified fixes pass full/piecewise graph capture, text/vision smoke, uncached 32k prefill and sustained C8/C16 decode in image `sha256:54d18cb8e1008e577f7daaa1e7d5b4689139de9402cb23dc76a93e7eefb48a8d`. The clean merge with #769/#770/#772 additionally passes DFlash graph capture, 15 complete Sieve requests and 30 seconds of uncached 32k prefill in image `sha256:1af7b5883d4da8cb4b8028bcabfa0e0054f783c6d3878650aee8274f033cb4ae`. These are composed serving checks, not isolated performance gains or native FP8/GB10 qualification. Matched measurements are in #772.

Changed-file checks pass except two inherited mypy findings: `_plans` and `_plan_caps` are annotated as `object` while existing code accesses `scratch_specs` and `cache_record_bytes`. No new type-check failure is introduced. Publication adds an explicit missing-API error and repository formatting to the measured source.

AI-assisted implementation and validation; human review requested.

### CODE REVIEW

**Comment by voipmonitor** [comment]
@coderabbitai review

**Comment by voipmonitor** [comment]
@coderabbitai review

Sparse-MLA preparation remains in this PR. Anonymous indexer identity handling and its regression test are in #772 alongside explicit prefixes, so the source branches now merge without manual conflict resolution. The PR body records the composed DFlash qualification and the dependency on B12X #379.
